src/recordlinker/linking/link.py

```diff
@@ -235,13 +249,13 @@ def link_record_against_mpi(
             # initialize a dictionary to hold the clusters of patients for each person
             clusters: dict[models.Person, list[models.Patient]] = collections.defaultdict(list)
 
-            # block on the pii_record and the algorithm's blocking criteria, then
+            # block on the cleaned_record and the algorithm's blocking criteria, then
             # iterate over the patients, grouping them by person
             with TRACER.start_as_current_span("link.block"):
                 # get all candidate Patient records identified in blocking
                 # and the remaining Patient records in their Person clusters
                 pats = mpi_service.BlockData.get(
-                    session, record, algorithm_pass, max_missing_allowed_proportion
+                    session, cleaned_record, algorithm_pass, max_missing_allowed_proportion
                 )
                 for pat in pats:
```

Comment:
You could also move the removal of skip values here instead of line 269/270 as I suggested before.